core/modules/wdb/QueryRunner.cc

```diff
@@ -254,6 +261,21 @@ bool QueryRunner::_fillRows(MYSQL_RES* result, int numFields, uint& rowCount, si
         unsigned int szLimit = std::min(proto::ProtoHeaderWrap::PROTOBUFFER_DESIRED_LIMIT,
                                         proto::ProtoHeaderWrap::PROTOBUFFER_HARD_LIMIT);
 
+        // for some to finish, otherwise
+        if (not _removedFromThreadPool) {
+            // This query has been answered by the database and the
+            // scheduler for this worker should stop waiting for it.
+            // leavePool() will tell the scheduler this task is finished
+            // and create a new thread in the pool to replace this one.
+            auto pet = _task->getAndNullPoolEventThread();
+            _removedFromThreadPool = true;
+            if (pet != nullptr) {
+                pet->leavePool();
+            } else {
+                LOGS(_log, LOG_LVL_WARN, "Result PoolEventThread was null. Probably already moved.");
```

Comment:
Is this a bug or a feature of the new code? If the former then the process need to be terminated right away. Otherwise these warnings will be buried in the long files.

Reply:
This is old code but there are other process that can move the thread out of the pool. The primary one being moved to the snail scheduler for taking too long.